worker/apicaller/connect.go

```diff
@@ -51,6 +51,10 @@ func OnlyConnect(a agent.Agent, apiOpen api.OpenFunc) (api.Connection, error) {
 	if !ok {
 		return nil, errors.New("API info not available")
 	}
+	// If we're a controller we have to use only first IP from APIInfo - localhost.
+	if _, isController := agentConfig.StateServingInfo(); isController {
+		info.Addrs = info.Addrs[:1]
```

Comment:
This seems odd to me. Where's the contract that states that the first address in the API info is always the localhost address? If controllers must always connect only to localhost, wouldn't a better solution be to make sure that the API info in the agent config only contains localhost addresses in the first place?